### ISSUE TITLE
Log messages targeting WebService not logged

### ISSUE THREAD

**Issue by XenonAtom** [NLog/NLog#2559]
**Type** (choose one): Question/Bug?

**NLog version**: 4.5.0-rc04

**Platform**: .NET Core 

**Current NLog config** (xml or C#, if relevant)

```xml
<?xml version="1.0" encoding="utf-8" ?>
<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd"
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
      xsi:schemaLocation="http://www.nlog-project.org/schemas/NLog.xsd NLog.xsd"
      autoReload="true"
      throwExceptions="false"
      internalLogLevel="Debug" internalLogFile="c:\temp\nlog-internal.log" >


  <extensions>
    <add assembly="NLog.Web"/>
    <add assembly="NLog.Web.AspNetCore" />
  </extensions>

  <targets>
    <target xsi:type="File" name="f" fileName="${basedir}/logs/${shortdate}.log"
            layout="${longdate} ${logger} ${uppercase:${level}} ${message} ${exception:format=toString}" />
    <!--<target xsi:type="File" name="ws" fileName="${basedir}/logs/${shortdate}-ws.log" layout="${longdate} ${logger} ${uppercase:${level}} ${message} ${exception:format=toString}" />-->
    <target xsi:type="WebService" name="ws" url="http://localhost:8088/api" protocol="HttpPost" encoding="UTF-8">
      <parameter name="Message" type="System.String" layout="${longdate} ${logger} ${uppercase:${level}} ${message} ${exception:format=toString}" />
    </target>
  </targets>

  <rules>
    <logger name="QueueSubscriber.WebLogger" minLevel="Info" writeTo="ws" final="true" />
    <logger name="*" minlevel="Trace" writeTo="f" />
  </rules>
</nlog>

```

Code to log error message:
```
var loggerFactory = new LoggerFactory().AddNLog();
      loggerFactory.ConfigureNLog("nlog.config");
      KeyValuePair<string, Exception> x = new KeyValuePair<string, Exception>("Message", e);
      List<KeyValuePair<string, Exception>> l = new List<KeyValuePair<string, Exception>>();
      l.Add(x);
      var log = loggerFactory.CreateLogger<QueueSubscriber.WebLogger>();
      log.LogError(msg, l);
```

**WHAT'S HAPPENING:**
Log messages from outside QueueSubscriber.WebLogger get logged to f. Log messages within QueueSubscriber.WebLogger don't get logged anywhere and Fiddler shows no traffic/attempt to log to webservice. No error is thrown; the log message seems to just get swallowed silently.

NLog internal log file with log level set to Debug:
[nlog-internal.log](https://github.com/NLog/NLog/files/1690909/nlog-internal.log)

Side note: I see the "For .NET Core users, please check the platform support: https://github.com/NLog/NLog/wiki/platform-support" message, but that page does not include .NET Core anywhere. WebService is listed as being supported on all (listed) platforms, though, so I believe it should work?

**Comment by snakefoot** [comment]
And you are logging messages with `Info`-Level to the `QS.WebLogger` ?

**Comment by snakefoot** [comment]
Funny that you have this line in your `nlog-internal.log`;

> 2018-02-02 09:56:05.6981 Debug Finished closing logging configuration.

Looks like you are changing the configuration multiple time.

**Comment by XenonAtom** [comment]
Yes, I'm only logging to ws at the Error level currently.

Snippet from code:

```
      var loggerFactory = new LoggerFactory().AddNLog();
      loggerFactory.ConfigureNLog("nlog.config");
      var log = loggerFactory.CreateLogger<QS.WebLogger>();
      log.LogError(msg);
```

I've tried a few ways of doing this (such as passing an args parameter to LogError containing the Message parameter), but they all have the same result.

I am creating multiple loggers (one for the standard logging to f and and for the webservice logging to ws); could that be why that line is showing up in the nlog-internal.log? I don't think the code changes the config anywhere

**Comment by snakefoot** [comment]
@XenonAtom Noticed that in the nlog-internal.log it is called `QueueSubscriber.WebLogger`. Are the logger-name in code and in nlog.config matching?

**Comment by XenonAtom** [comment]
Ah, yeah, they match. Sorry, I went a bit overboard trying to genericize the code and shortened it to QS in the parts I was posting without thinking about it getting printed to the nlog's internal log

The file-target is getting written to by a logger created in the same fashion, but class is .Listener instead of .WebLogger

**Comment by snakefoot** [comment]
To repeat my questions:

> What ${logger} is written to the file-target?

> Are the logger-name in code and in nlog.config matching?

**Comment by XenonAtom** [comment]
> Are the logger-name in code and in nlog.config matching? (For QS-Errors)
Yes, they match

> What ${logger} is written to the file-target? (For QS-Errors)
QueueSubscriber.Listener

**Comment by snakefoot** [comment]
Could you update the `Nlog.config` posted, along with creating a new `nlog-internal.log` ?

Maybe you should test your filtering with an extra file-target, and when that works, then point it to the webservice-target.

**Comment by snakefoot** [comment]
@XenonAtom Can you remove the `final="true` from the 'QueueSubscriber.WebLogger'-logging-rule, and tell what is written to the file when doing `LogError`

**Comment by XenonAtom** [comment]
I've updated the first post's Nlog.config and replaced the nlog-internal.log with a new one, as well as moving from LogError to just Log.

Having ws target be a file works fine (I left that version of the target in the config commented out to show how it was set up).

Here's an example of what gets written to the file log for a LogError:
`2018-02-02 14:39:23.4525 QueueSubscriber.WebLogger ERROR Error for web logging` 

I note that this doesn't include the exception message ("Forced exception"), probably because I'm passing it as a parameter, which the target isn't set up for. I need to shift focus for a bit, but when I come back to this I'll look into things like making sure that CaptureMessageTemplates isn't getting set to false or anything like that

**Comment by snakefoot** [comment]
Think the solution is to modify the WebService-config to include  `proxyType="NoProxy"`:

`<target xsi:type="WebService" name="ws" url="http://localhost:8088/api" protocol="HttpPost" encoding="UTF-8" proxyType="NoProxy">`

Will see if we can improve the out-of-the-box experience on NetCore.

**Comment by snakefoot** [comment]
Created #2563

**Comment by snakefoot** [comment]
Could also be that your application exits before the async WebRequest is performed, so maybe have this line before application-exit:

`NLog.LogManager.Flush();`

**Comment by TimCulhaneSGSI** [comment]
So it turns out that my issue was a misunderstanding and not realizing that Fiddler won't (by default, at least) show traffic to localhost. Once I directed the traffic elsewhere, the message popped up like expected. (I did have the noProxy thing added, which may have also made the difference).

Thanks for the help, and glad I at least helped you stumble into an improvement for NLog!

**Comment by XenonAtom** [comment]
Turns out that my issue was that I just didn't realize that Fiddler won't (by default, at least) show traffic to localhost; once I gave it a different target, the message showed up as expected.

Thanks for looking into this and helping!

**Comment by snakefoot** [comment]
@XenonAtom Thank you for reporting back. At least it allowed us to remove an internal exception everything doing a WebRequest on NetCore (Will be fixed in NLog 4.5-rc05)
